articles/nginx-reverse-proxy/index.md

```diff
@@ -0,0 +1,192 @@
+
+The general DNS Configurations would be something like
+- Server app mapped to the server.domain
+- Client app mapped to the client.domain
+
+My Localhost Config, in this case, would be
+- Server mapped to server.localhost
+- Client mapped to client.localhost
+
+There are two standard protocols HTTP and HTTPS. The default port for HTTP is 80 and HTTPS is 443. This is the reason we must not run our applications on these ports because our NGINX server would be running on these two ports. All the requests the client makes would either be redirected to port 80 or 443 from where it would be redirected internally to the corresponding application.
+
+### Step 3 - Configure NGINX at 80 for HTTP and 443 for HTTPS
+Now that we have our apps running and our DNS records ready. We can start configuring our NGINX Reverse Proxy to make it all work.  
+
+By default, the configuration file is named nginx.conf and placed in the directory /usr/local/nginx/conf, /etc/nginx, or /usr/local/etc/nginx for Linux and Debian Based systems.
+
+On Windows, the file is placed inside the installation folder, nginx/conf/nginx.conf.
+
+Add these configurations inside the HTTP block.
+
+#### Step 3.1 - HTTP
+
+```
+server {       
+     listen       80;
+     server_name  server.domain;
+     location / {
+          proxy_pass "http://localhost:3000" ;
+     }
+}
+server {       
+     listen       80;
+     server_name  client.domain;
+
+     location / {
+          proxy_pass "http://localhost:3001" ;
+     }
+}
+```
+
+#### Step 3.2 - HTTPS
+
+```
+server {
+    listen 443 ssl http2;
+    listen [::]:443 ssl http2;
+
+    server_name server.domain;
+
+    location / {    
+        proxy_pass "http://localhost:3000/";
+    }
+
+    ssl_certificate <location of SSL certificate>;
+    ssl_certificate_key <location of SSL certificate Key>;
+}
+server {
+    listen 443 ssl http2;
+    listen [::]:443 ssl http2;
+
+    server_name client.domain;
+
+    location / {    
+        proxy_pass "http://localhost:3001/";
+    }
+
+    ssl_certificate <location of SSL certificate>;
+    ssl_certificate_key <location of SSL certificate Key>;
+}
+```
+
+**NOTE**
+These are the basic minimum configurations required to successfully configure NGINX for reverse proxying. Feel free to explore other config parameters as well.
+
+Make sure to change the domain name to your domain. For a SSL Certificate and Key, you can obtain them from your SSL provider.
+
+If you don't have one, Use this free service [LetsEncrypt](https://letsencrypt.org/).
+
+Follow their documentation to get Free SSL instantly!
+
+### Step 4 - Save and Restart
+After editing, save your changes. Use the `nginx -t` command to test your changes before actually reloading NGINX. It is good practice do this to make sure your server won't crash if there were any errors in your config file.
+
+Once you get a message that the test is successful, you can go ahead and restart NGINX.  
```

Comment:
You should add the command to restart NGINX. Readers may not know what it is.

Reply:
Thanks and Done 👍